src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
It is known on the client but it is not known on the server - the server picks the protocol upon receiving negotiate from the client. (i.e. the server cannot use _options.WebSocketMessageType here: https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs#L183 because it was set before negotiation and therefore no-one knew what protocol the client will request).